components/infra-proxy-service/server/nodes.go

```diff
@@ -12,14 +12,38 @@ import (
 	"github.com/chef/automate/components/infra-proxy-service/validation"
 )
 
+// GetNodes fetches the nodes from chef infra server
+func (s *Server) GetNodes(ctx context.Context, req *request.Nodes) (*response.Nodes, error) {
+	c, err := s.createClient(ctx, req.OrgId, req.ServerId)
+	if err != nil {
+		return nil, err
```

Comment:
I wonder if we want to wrap these errors a bit to make sure we return grpc Status errors: https://pkg.go.dev/google.golang.org/grpc/status

Reply:
Yeah, I have moved wrapping in `proxy.go#createClient` with status.
here a bit of generalization

```
func ParseAPIError(err error) error {
	chefError, _ := chef.ChefError(err)
	if chefError != nil {
		switch chefError.StatusCode() {
		case http.StatusBadRequest:
			return status.Errorf(codes.InvalidArgument, chefError.StatusMsg())
		case http.StatusUnauthorized:
			return status.Errorf(codes.Unauthenticated, chefError.StatusMsg())
		case http.StatusForbidden:
			return status.Errorf(codes.PermissionDenied, chefError.StatusMsg())
		case http.StatusConflict:
			return status.Errorf(codes.AlreadyExists, chefError.StatusMsg())
		case http.StatusNotFound:
			return status.Errorf(codes.NotFound, chefError.StatusMsg())
		default:
			return status.Error(codes.InvalidArgument, chefError.StatusMsg())
		}
	}
```